### PR TITLE
ARGO-136 Multitenancy for site availability

### PR DESCRIPTION
**Pull request by makistsantekidis**
* View changed to group by endpoint groups
* query parameters group_name, supergroup_name, job, start_time, end_time
* Accept an additional http header in the request (api_key)
* Authenticate the api_key, find the relevant tenant and return the tenant db conf
* use the tenant db conf in the subsequent queries and operations instead of the main db conf

### CODE REVIEW

**Comment by makistsantekidis** [comment]
please review

/cc @pkoro @dpavlos @kaggis

**Comment by dpavlos** [comment]
+1 (just remove the deprecated code)

File: app/endpointGroupAvailability/endpointGroupAvailabilityController.go [hunk 1]
```diff
@@ -24,15 +24,17 @@
...
+
+	"github.com/argoeu/argo-web-api/utils/authentication"
 	"github.com/argoeu/argo-web-api/utils/caches"
 	"github.com/argoeu/argo-web-api/utils/config"
 	"github.com/argoeu/argo-web-api/utils/mongo"
-	"net/http"
-	"strings"
 )
 
 func List(r *http.Request, cfg config.Config) (int, http.Header, []byte, error) {
```
Comment (dpavlos) [inline app/endpointGroupAvailability/endpointGroupAvailabilityController.go]: godoc here

File: app/endpointGroupAvailability/endpointGroupAvailabilityModel.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+import (
+	"encoding/xml"
+	"strconv"
+	"time"
+
+	"labix.org/v2/mgo/bson"
+)
+
+type MongoInterface struct {
```
Comment (dpavlos) [inline app/endpointGroupAvailability/endpointGroupAvailabilityModel.go]: Please add godoc for exported structs.

**Comment by dpavlos** [comment]
+1

**Pull request merged**